app-backend/tool/src/main/scala/ast/MapAlgebraAST.scala

```diff
@@ -58,22 +57,17 @@ object MapAlgebraAST {
       extends Operation("classify")
 
   /** Map Algebra sources (leaves) */
-  sealed abstract class Source[+T](val `type`: String) extends MapAlgebraAST {
-    def value: Option[T]
+  @JsonCodec
+  case class Source(id: UUID, label: Option[String]) extends MapAlgebraAST {
```

Comment:
Ah right, since an AST (a "Tool") might be passed around and fulfilled in multiple ways.